### PR TITLE
Augment python interface of Point

### PR DESCRIPTION
**Pull request by Rombur**

### CODE REVIEW

File: contrib/python-bindings/include/point_wrapper.h [hunk 1]
```diff
@@ -51,12 +56,89 @@ namespace python
...
 
     /**
+     * Return the Euclidean distance of this point to the point p, i.e., the
+     * l2_norm of the difference between the vectors representing the two
+     * points.
+     */
+    double distance(const PointWrapper &p) const;
+
+    /**
+     * Return the l2_norm of the vector.
```
Comment (davydden) [inline contrib/python-bindings/include/point_wrapper.h]: maybe rephrase along the lines that this is a norm of the vector connecting origin to the given point.

File: contrib/python-bindings/include/point_wrapper.h [hunk 2]
```diff
@@ -51,12 +56,89 @@ namespace python
...
+     */
+    bool operator!= (const PointWrapper &p) const; 
+
+    /**
+     * Test for equality of two tensors.
+     */
+    bool operator== (const PointWrapper &p) const;
+
+    /**
+     * Return the scalar product of the vectors representing two points.
```
Comment (davydden) [inline contrib/python-bindings/include/point_wrapper.h]: I guess it's trivial to add a TensorWrapper class and make PointWrapper derive from it...

I don't mind having a bit of inconsistency here, at least for now.

File: contrib/python-bindings/include/point_wrapper.h [hunk 3]
```diff
@@ -51,12 +56,89 @@ namespace python
...
+     */
+    PointWrapper operator-(const PointWrapper &) const;
+
+    /**
+     * The opposite point.
+     */
+    PointWrapper operator-() const;
+
+    /**
+     * Divide the current point by a factor.
```
Comment (davydden) [inline contrib/python-bindings/include/point_wrapper.h]: divide coordinates of the... ? (same below for similar operations)

File: contrib/python-bindings/include/point_wrapper.h (hunk 2 above)
Comment (Rombur) [inline contrib/python-bindings/include/point_wrapper.h]: Yes, we could add a TensorWrapper but unless we want to have a complete python interface, I don't see the point of adding it.

**Comment by Rombur** [comment]
I have made the changes.

File: contrib/python-bindings/include/point_wrapper.h (hunk 2 above)
Comment (davydden) [inline contrib/python-bindings/include/point_wrapper.h]: i wanted to say "is NOT trivial", so yes, i agree with your point
